boa/src/builtins/function/arguments.rs

```diff
@@ -160,87 +161,83 @@ impl Arguments {
         .expect("DefinePropertyOrThrow must not fail per the spec");
 
         // 17. Let mappedNames be a new empty List.
-        // using a set to optimize `contains`
-        let mut mapped_names = FxHashSet::default();
-
         // 12. Let parameterNames be the BoundNames of formals.
         // 13. Let numberOfParameters be the number of elements in parameterNames.
         // 18. Set index to numberOfParameters - 1.
         // 19. Repeat, while index ≥ 0,
         // a. Let name be parameterNames[index].
```

Comment:
Could this somehow be mapped to the code? I guess we are changing a bit the algorithm here for performance, right?

Reply:
Not really performance. The spec works with identifiers (`name`) but we do not have identifiers at runtime anymore. We have to work with the binding indices on the environments. The argument bindings are always the first few binding in the environment and we use that to decide what indices the getters and setters should have here.

For example:
```js
function f(a, b, c) {}
f(1,2,3)
```
Looks like this in the function environment:
| binding index | `arguments` property | identifier | value |
|---|---|---|---|
| 0 | arguments[0] | a | 1 |
| 1 | arguments[1] | b | 2 |
| 2 | arguments[2] | c | 3 |

This is pretty straightforward, but we can also have this:
```js
function f(a, a, c, c) {}
f(1,2,3)
```
Which translates to:
| binding index | `arguments` property | identifier | value |
|---|---|---|---|
| - | arguments[0] | - | - |
| 0 | arguments[1] | a | 2 |
| - | arguments[2] | - | - |
| 1 | arguments[3] | c | undefined |

As you can see, we have to work out the binding index based on the number of arguments and based on the parameter names.

I think I will add this comment or something similar as documentation to the code to explain how this is different from the spec.

Reply:
That would be great, yes!!

Reply:
Added!